modules/ScreenNameLocalFormatter/src/test/java/com/liferay/formatter/generator/service/test/NewScreenNameGeneratorTest.java

```diff
@@ -0,0 +1,236 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.formatter.generator.service.test;
+
+import com.liferay.formatter.generator.service.NewScreenNameGenerator;
+import com.liferay.formatter.keys.NewFormatterKeys;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.model.GroupWrapper;
+import com.liferay.portal.kernel.model.UserWrapper;
+import com.liferay.portal.kernel.service.GroupLocalService;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.PrefsProps;
+import com.liferay.portal.kernel.util.Props;
+import com.liferay.portal.kernel.util.PropsKeys;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.mockito.AdditionalMatchers;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ * @author leonardo.ferreira
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class NewScreenNameGeneratorTest {
+
+	@Test
+	public void testGenerate_AddsEmailSufix_When_ScreenNameIsUnusedByOtherUserOrGroup_And_ScreenNameDoesNotContainsReservedWords()
+		throws Exception {
+
+		// Arrange
+
+		String input = "user@liferay.com";
+		String expected = "user" + _correctEmail;
+		Mockito.when(
```

Comment:
It seems you have some duplicated setup done in all your tests.
This line for example

Create a setup method annotated with `@Before`. This will run before every test method.
All common setup should be moved to this method